flow/layers/display_list_layer.cc

```diff
@@ -91,24 +91,48 @@ bool DisplayListLayer::Compare(DiffContext::Statistics& statistics,
 void DisplayListLayer::Preroll(PrerollContext* context,
                                const SkMatrix& matrix) {
   TRACE_EVENT0("flutter", "DisplayListLayer::Preroll");
+  context->raster_cached_entries.emplace_back(
+      RasterCacheEntry(this, *context, matrix));
```

Comment:
DisplayList objects should not be cached by layer. They can move around the tree and sometimes be duplicated and neither of those practices should break their caching. The DisplayList cache is a different cache than the layer cache and so needs to be treated differently as a RasterCacheEntry. For instance, there are 3 different RasterCache::Prepare calls and 3 different RasterCache::Rasterize[Layer, Picture, DisplayList] calls.

It would be great to consolidate those three sets of caching methods into a single path, but it cannot be done by caching DisplayList objects by their layer.

Reply:
I have distinguished the `Layer`, `DisplayList`, `SkPicture`, Please give me a review. Thank you @flar